src/Microsoft.IdentityModel.Clients.ActiveDirectory/Internal/Instance/Authenticator.cs

```diff
@@ -90,26 +88,29 @@ public Authenticator(string authority, bool validateAuthority)
 
         public string SelfSignedJwtAudience { get; private set; }
 
+        public string PreferredCache { get; private set; }
```

Comment:
we can always ask the cache for the value. why are PreferredCache and Aliases needed here?

Reply:
+1

These values should be pulled from cache on use. There are flows where the authority url can change mid flow. It's better to not have to code around that by just asking the authority cache directly.